react/src/base/inputs/SprkTextInput/SprkTextInput.js

```diff
@@ -67,7 +67,7 @@ class SprkTextInput extends Component {
               forwardedRef={forwardedRef}
               type={type}
               formatter={formatter}
-              errorContainerId={errorContainerId}
+              errorContainerId={valid ? undefined : errorContainerId}
```

Comment:
Nice! This is a good way to make sure screen readers aren't confused if the error doesn't exist.